lib/iris/fileformats/grib/_save_rules.py

```diff
@@ -164,9 +164,16 @@ def identification(cube, grib):
     # operational product, operational test, research product, etc
     # (missing for now)
     gribapi.grib_set_long(grib, "productionStatusOfProcessedData", 255)
+
+    # Code table 1.4
     # analysis, forecast, processed satellite, processed radar,
-    # (analysis and forecast products for now)
-    gribapi.grib_set_long(grib, "typeOfProcessedData", 2)
+    if cube.coords('realization') and len(cube.coords('realization')) == 1:
+        if cube.coord('realization').points != (0,):
+            gribapi.grib_set_long(grib, "typeOfProcessedData", 4)
+        else:
+            gribapi.grib_set_long(grib, "typeOfProcessedData", 3)
```

Comment:
@marqh Fancy adding a minor comment to explain the meaning of the `typeOfProcessedData` values ... or simply reference the GRIB code table, now that would be handy? I'm guessing it's **Code table 4.3** ?